### PR TITLE
[TIMOB-20283] Fix for FB crash on iOS 8.x

### PR DESCRIPTION
**Pull request by boarnoah**
Quick fix for [TIMOB-20283](https://jira.appcelerator.org/browse/TIMOB-20283).
I haven't messed with Obj-C before so if someone could verify I didn't inadvertently break something major it would be great (I tested this locally on various devices at the office without issues (ios 9 and 8)).
Thanks

### CODE REVIEW

**Comment by boarnoah** [comment]
Getting 504 for the Appcelerator-CLA page, if someone else could duplicate and PR I don't mind.

**Comment by boarnoah** [comment]
I got the idea from http://stackoverflow.com/questions/31504765/crash-in-openurl-possible-facebook-sdk-bug-in-ios-9.

It shouldn't affect iOS 8 if it's null, and as far as I can see iOS 9.x is happy with it being a string. It's probably not ideal like you mentioned though.

**Comment by annakozy2011** [comment]
I am concerned about referrals. I am not sure what FB sends in this annotation parameter, when you install from the paid campaign on FB, and for my client it´s of vital importance. 
Thankfully, the number of 8.x systems is small among our users and diminishing, so we´ll only lose some small share of information in FB statistics, if I send nil ONLY FOR 8.X systems. 

It changes though if I start sending nil for all the crowd. .... and in another 3-4 months, 8.x will probably disappear from the horizon, and we´ll be able for forget about this quick fix entirely. 

My client is a startup, and I suspect that their investors look at their FB and other statistics, to determine their health :D So no statistics - no money :D

**Comment by boarnoah** [comment]
@annakozy2011 I'm implementing the fix in the way you recommended in the appcelerator jira. 

We are seeing pretty much the same trend on our iOS 8 userbase (iOS userbase ~80k) too, it's just ideal if we could include this as a fix till the day its fully phased out.

File: ios/Classes/FacebookModule.m [hunk 1]
```diff
@@ -48,7 +48,14 @@ -(BOOL)handleRelaunch
     launchOptions = [appDelegate launchOptions];
     NSString *urlString = [launchOptions objectForKey:@"url"];
     NSString *sourceApplication = [launchOptions objectForKey:@"source"];
-    NSString *annotation = [launchOptions objectForKey:UIApplicationOpenURLOptionsAnnotationKey];
+    NSString *annotation;
+    
+    if (SYSTEM_VERSION_GREATER_THAN_OR_EQUAL_TO(@"9.0")) {
+        annotation = [launchOptions objectForKey:UIApplicationOpenURLOptionsAnnotationKey];
```
Comment (hansemannn) [inline ios/Classes/FacebookModule.m:54]: Please use the builtin helper `[TiUtils isiOS9OrGreater]` instead of a custom macro, thanks!

**Comment by hansemannn** [comment]
Reviewed the PR, please address the iOS9-check comment and make sure to sign the CLA. 

As there are currently problems with the CLA server, use the following hook: Browse in private mode with your browser (Chrome: CMD+Shift+N) and try it again. Thank you!

**Comment by boarnoah** [comment]
Working on it right now.
I didn't have luck using private mode in chrome+firefox. Still get 504 gateway issues. If they don't fix themselves soon enough, could you post the final version of the PR?

**Comment by hansemannn** [comment]
Yes, I could but I don't want to steal your credits on contribution :-)

**Comment by hansemannn** [comment]
Closing in favor of cherry-picked https://github.com/appcelerator-modules/ti.facebook/pull/42, thank you!

**Pull request closed without merging**